neqo-transport/src/connection/mod.rs

```diff
@@ -2846,32 +2826,23 @@ impl Connection {
                         .local
                         .get_integer(tparams::INITIAL_MAX_STREAM_DATA_BIDI_REMOTE)
                 } else {
-                    if stream_idx > self.indexes.local_max_stream_uni {
-                        qwarn!(
-                            [self],
-                            "remote uni stream create blocked, next={:?} max={:?}",
-                            stream_idx,
-                            self.indexes.local_max_stream_uni
-                        );
-                        return Err(Error::StreamLimitError);
-                    }
                     self.tps
                         .borrow()
                         .local
                         .get_integer(tparams::INITIAL_MAX_STREAM_DATA_UNI)
                 };
 
-                loop {
-                    let next_stream_id =
-                        next_stream_idx.to_stream_id(stream_id.stream_type(), stream_id.role());
+                while self.remote_streams_fc[stream_id.stream_type()].is_new_stream(stream_id)? {
+                    let next_stream_id = self.remote_streams_fc[stream_id.stream_type()]
+                        .add_stream()
```

Comment:
I think that you can collapse these two calls into one, like:

```rust
while self.stream_limits.remote_mut(stream_id.stream_type()).add_stream(stream_id)? {
  // ...
}
```

It's probably not so terrible here to get the transport parameter value more than once.  Opening multiple streams is rare enough that the extra calls won't hurt.